include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -424,14 +410,17 @@ class transform_iterator
     typedef typename ::std::iterator_traits<_Iter>::pointer pointer;
     typedef typename ::std::iterator_traits<_Iter>::iterator_category iterator_category;
 
-    transform_iterator(_Iter __it = _Iter(), _UnaryFunc __unary_func = _UnaryFunc())
-        : __my_it_(__it), __my_unary_func_(__unary_func)
-    {
-    }
+    transform_iterator() = default;
+    transform_iterator(_Iter __it) : __my_it_(__it), __my_unary_func_() {}
+    transform_iterator(_Iter __it, _UnaryFunc __unary_func) : __my_it_(__it), __my_unary_func_(__unary_func) {}
```

Comment:
It seems that `transform_iterator() = default;` is inherently different from the previous default constructor with default param arguments, and it is causing failures in `iterators.pass`.

Reply:
Reverting "extra" changes which are not toward this PR's goal of fixing device copyable for fancy iterators.  (default constructor changes, etc.  
This makes the above comment unrelated to this PR.